src/vellum/workflows/nodes/displayable/bases/api_node/node.py

```diff
@@ -52,6 +52,9 @@ def _run(
         headers: Any = None,
         bearer_token: Optional[VellumSecret] = None,
     ) -> Outputs:
+        if not url or not isinstance(url, str) or not url.strip():
+            raise NodeException("URL is required and must be a non-empty string", code=WorkflowErrorCode.INVALID_INPUTS)
```

Comment:
after the feedback above, you should be able to just do `self._validate()` here. The other two helper methods below shouldn't need to since we'll already be valid